File: Day 20 : Sorting/Solution.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int i, j, n, temp;
	scanf("%d", &n);

    int* a = malloc(n * sizeof(int));

    for (int i = 0; i < n; i++) {
        int a_item;
		scanf("%d", &a_item);

        *(a + i) = a_item;
    }

    int numberOfSwaps = 0;
    for(i = 0; i < n; i++)
    {   
        for(j = 0; j < n - 1; j++)
        {
            if(a[j] > a[j + 1])
            {
                temp = a[j];
                a[j] = a[j + 1];
                a[j + 1] = temp;
                numberOfSwaps++;
            }
        }
        
        if(numberOfSwaps == 0)
            break;
    }
    
    printf("Array is sorted in %d swaps.\n",numberOfSwaps);
    printf("First Element: %d\n", a[0]);
    printf("Last Element: %d\n", a[n-1]);

    return 0;
}

File: Day 20 : Sorting/Solution.java
import java.io.*;
import java.util.*;

public class Solution
{
    public static void main(String[] args) throws IOException
    {
        Scanner scan = new Scanner(System.in);
        
        int numberOfSwaps = 0;
        List<Integer> a = new ArrayList<Integer>();

        int n = scan.nextInt();

        for(int i = 0; i < n; i++)
        {
            a.add(i, scan.nextInt());
        }
        
        for(int i = 0; i < n; i++)
        {
            for(int j = 0; j < n-1; j++)
            {
                if(a.get(j) > a.get(j+1))
                {
                    Collections.swap(a, j, j+1);
                    numberOfSwaps++;
                }
            }
            if(numberOfSwaps == 0)
            {
                break;
            }
        }
        
        System.out.println("Array is sorted in " + numberOfSwaps + " swaps.");
        System.out.println("First Element: " + a.get(0));
        System.out.println("Last Element: " + a.get(n-1));
        
        scan.close();
    }
}
